moped-database/migrations/1704744986000_substantial_completion_date/up.sql

```diff
@@ -0,0 +1,450 @@
+      FROM moped_project
+      WHERE project_id = mp.parent_project_id
+    ) as parent_project_name,
+    cpl.children_project_ids,
+    string_agg(DISTINCT me2.entity_name, ', '::text) AS project_partner,
+    (SELECT JSON_AGG(json_build_object('signal_id', feature_signals.signal_id, 'knack_id', feature_signals.knack_id, 'location_name', feature_signals.location_name, 'signal_type', feature_signals.signal_type, 'id', feature_signals.id))
+        FROM moped_proj_components components   
+        LEFT JOIN feature_signals
+          ON (feature_signals.component_id = components.project_component_id)
+        WHERE TRUE
+          AND components.is_deleted = false
+          AND components.project_id = mp.project_id
+          AND feature_signals.signal_id is not null
+          AND feature_signals.is_deleted = false
+        ) as project_feature,
+    fsl.funding_source_name,
+    ptl.type_name,
+    ( -- get the date of the construction phase with the earliest start date
+      SELECT min(phases.phase_start)
+      FROM moped_proj_phases phases
+      WHERE true
+        AND phases.project_id = mp.project_id 
+        AND phases.phase_id = 9 -- phase_id 9 is construction
+        AND phases.is_deleted = false
+    ) AS construction_start_date,
+    ( -- get the date of the completion phase with the latest end date
+      SELECT max(phases.phase_end)
+      FROM moped_proj_phases phases
+      WHERE true 
+        AND phases.project_id = mp.project_id 
+        AND phases.phase_id = 11 -- phase_id 11 is complete
+        AND phases.is_deleted = false
+      ) AS completion_end_date,
+    ( -- get the earliest confirmed phase_start or phase_end with a simple phase of 'Complete' 
+      SELECT
+          min(min_confirmed_date)
+      FROM (
+          -- earliest confirmed phase start
+          SELECT
+              min(phases.phase_start) AS min_confirmed_date
+          FROM
+              moped_proj_phases phases
+          LEFT JOIN moped_phases ON phases.phase_id = moped_phases.phase_id
+          WHERE
+              TRUE
+              AND phases.phase_start IS NOT NULL
+              AND phases.is_phase_start_confirmed = TRUE
+              AND phases.project_id = mp.project_id
+              AND moped_phases.phase_name_simple = 'Complete'
+              AND phases.is_deleted = FALSE
+          UNION ALL
+          -- earliest confirmed phase end
+          SELECT
+              min(phases.phase_end) AS min_confirmed_date
+          FROM
+              moped_proj_phases phases
+          LEFT JOIN moped_phases ON phases.phase_id = moped_phases.phase_id
+          WHERE
+              TRUE
+              AND phases.phase_end IS NOT NULL
+              AND phases.is_phase_end_confirmed = TRUE
+              AND phases.project_id = mp.project_id
+              AND moped_phases.phase_name_simple = 'Complete'
+              AND phases.is_deleted = FALSE
+        ) min_confirmed_dates
+      ) AS substantial_completion_date,
```

Comment:
This SQL was tricky for me. We're trying to find the earliest `phase_start` date + `is_start_confirmed = true` OR the earliest `phase_end` date + `is_end_confirmed = true`.

The easiest way I could think of was to use a `UNION`